### PR TITLE
Fixed address street missing in billing address for German orders.

### PR DESCRIPTION
**Pull request by creative-andrew**
### All Submissions:

* [x] Does your code follow the [Extendables](https://extendomattic.wordpress.com/standardizations/) standards?
* [x] Have you successfully run tests with your changes locally?

### Changes proposed in this Pull Request:

There is a special handling regarding the billing address for german orders that contemplates different scenarios.
However there is one missing that can lead to the billing address street to be empty.

If the buyer leaves `$address->AddressLine1` empty but fills `$address->AddressLine2`, no address will be saved. 

This is even how [Amazon in its documentation](https://amazonpaylegacyintegrationguide.s3.amazonaws.com/docs/eu/amazon-pay-automatic/sample-addresses-and-payment-instruments.html) passes the information. 

image

Notice how the street number won't appear in the billing section.
image

### How to test the changes in this Pull Request:

1. Create an Amazon Pay test account (Germany)
2. Make sure that address country code is DE
3. Create a billing address leaving the first field empty, but fill the second one.
4. Pay something.
5. Go to the the WooCommerce order page to check the billing address. 

### Other information:

* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

Fixed address street missing in billing address for German orders when $address->AddressLine1 is not present. 

Closes: https://github.com/woocommerce/woocommerce-gateway-amazon-pay/issues/203

### CODE REVIEW

File: includes/class-wc-amazon-payments-advanced-api-abstract.php [hunk 1]
```diff
@@ -423,6 +423,10 @@ public static function format_address( $address ) {
 				$formatted['company']   = (string) $address->AddressLine1;
 				$formatted['address_1'] = (string) $address->AddressLine2;
 
+			} elseif ( ! empty( $address->AddressLine2 ) && empty( $address->AddressLine1 ) ) {
```
[the remark this answers is not in the document]
Comment (dpanta94) [reply includes/class-wc-amazon-payments-advanced-api-abstract.php:426]: Hey @kalessil, couldn't find anything reported on GitHub but asked the team responsible for handling support tickets via WooCommerce's Zendesk and there are reported issues relevant there.

[the remark this answers is not in the document]
Comment (kalessil) [reply includes/class-wc-amazon-payments-advanced-api-abstract.php:426]: Hey @geist-ahnen, can we revisit please this PR and move it ahead with the suggested changes?

[the remark this answers is not in the document]
Comment (geist-ahnen) [reply includes/class-wc-amazon-payments-advanced-api-abstract.php]: @kalessil I've refactored the code based on the suggested changes by @dmvrtx.

I've added a unit test to make sure that the functionality works as expected.

**Comment by geist-ahnen** [comment]
By the way, the link in the comment that explains why german addresses have that exception does not work: `// @see https://github.com/woothemes/woocommerce-gateway-amazon-payments-advanced/issues/25`

Why is that german addresses don't have address `$formatted['address_2']`?

**Comment by geist-ahnen** [comment]
Any news on this? @kalessil

**Review by dpanta94 — COMMENTED** [review]
Hey @creative-andrew, could you please merge the main branch into your branch (resolving conflicts) so we can create a build to QA your changes ? Please, also remove the tests since we are working on refactoring them.

**Comment by creative-andrew** [comment]
@dpanta94 I have merged the main branch into my branch (resolving conflicts) and I have also removed the tests as requested.

This is ready to be merged.

cc: @kalessil

**Comment by kalessil** [comment]
Thank you @creative-andrew ! @dpanta94 will be available in ~1 week, so if all going well we'll merge/release the fix in January.

**Review by kalessil — APPROVED** [review]
LGTM!

**Comment by kalessil** [comment]
@dpanta94: hading the PR over to you. Feel free to merge.

**Comment by dpanta94** [comment]
Thank you very much for your contribution @creative-andrew ! Merging!

**Pull request merged**